storage/beacondb/eth1.go

```diff
@@ -22,38 +22,32 @@ package beacondb
 
 import (
 	ctypes "github.com/berachain/beacon-kit/consensus-types/types"
-	"github.com/berachain/beacon-kit/primitives/bytes"
+	"github.com/berachain/beacon-kit/primitives/common"
+	"github.com/berachain/beacon-kit/primitives/version"
 )
 
 // GetLatestExecutionPayloadHeader retrieves the latest execution payload
 // header from the BeaconStore.
 func (kv *KVStore) GetLatestExecutionPayloadHeader() (
 	*ctypes.ExecutionPayloadHeader, error,
 ) {
-	// NOTE: unmarshalling this struct is NOT affected by it's own fork version. The versioned
-	// codec is left in for backwards compatibility.
-	forkVersion, err := kv.latestExecutionPayloadVersion.Get(kv.ctx)
-	if err != nil {
-		return nil, err
-	}
-	kv.latestExecutionPayloadCodec.SetActiveForkVersion(bytes.FromUint32(forkVersion))
 	return kv.latestExecutionPayloadHeader.Get(kv.ctx)
 }
 
 // SetLatestExecutionPayloadHeader sets the latest execution payload header in
 // the BeaconStore.
 func (kv *KVStore) SetLatestExecutionPayloadHeader(
-	payloadHeader *ctypes.ExecutionPayloadHeader,
+	payloadHeader *ctypes.ExecutionPayloadHeader, forkVersion common.Version,
```

Comment:
spitballing here: an alternative could be to have a `SetLatestExecutionPayloadHeaderPreElectra` and a `SetLatestExecutionPayloadHeader`.
From Pectra on we would call only the latter, but clients of this method would have to decide which method to call.
Probably does not make much of a different, just pointing it out